bin/node/browser-testing/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Running
+//! Running this test can be done with
+//! ```
+//! wasm-pack test --firefox --release --headless bin/node/browser-testing
+//! ```
+//! For debug infomation, such as the informant, run without the `--headless`
+//! flag and open a browser to the url that `wasm-pack test` outputs.
+//! For more infomation see https://rustwasm.github.io/docs/wasm-pack/.
+
+use wasm_bindgen_test::{wasm_bindgen_test, wasm_bindgen_test_configure};
+use wasm_bindgen_futures::JsFuture;
+use wasm_bindgen::JsValue;
+use wasm_bindgen::prelude::wasm_bindgen;
+use jsonrpc_core::types::{MethodCall, Success, Version, Params, Id};
+use serde::de::DeserializeOwned;
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+#[wasm_bindgen(module = "/src/ws.js")]
```

Comment:
Idea: what about we do it this way in the real browser node code as well?

Reply:
Yeah, that'd probably be better than copy-pasting it around. I'll look into that in a bit.

Reply:
See https://github.com/libp2p/rust-libp2p/pull/1454.